### ISSUE TITLE
react 16 warning: t.componentWillReceiveProps(): Assigning directly to this.state is deprecated (except inside a component's constructor). Use setState instead.

### ISSUE THREAD

**Issue by capaj** [mac-s-g/react-json-view#107]
```
bundle.js:2343 Warning: t.componentWillReceiveProps(): Assigning directly to this.state is deprecated (except inside a component's constructor). Use setState instead.
printWarning @ bundle.js:2343
warning @ bundle.js:2367
callComponentWillReceiveProps @ bundle.js:24918
updateClassInstance @ bundle.js:25090
updateClassComponent @ bundle.js:25340
beginWork @ bundle.js:25716
performUnitOfWork @ bundle.js:27672
workLoop @ bundle.js:27781
callCallback @ bundle.js:17021
invokeGuardedCallbackDev @ bundle.js:17060
invokeGuardedCallback @ bundle.js:16917
performWork @ bundle.js:27901
batchedUpdates @ bundle.js:28352
performFiberBatchedUpdates @ bundle.js:17368
stackBatchedUpdates @ bundle.js:17359
batchedUpdates @ bundle.js:17373
batchedUpdatesWithControlledComponents @ bundle.js:17386
dispatchEvent @ bundle.js:17596
```
I got this today with react 16 beta 5

**Comment by mac-s-g** [comment]
thanks for pointing this out.  I'll resolve as soon as I get a chance.

**Comment by mac-s-g** [comment]
I installed react and react-dom at `16.0.0-beta.5` and built my project.  I'm not seeing any warnings.

I tried running with both `process.env.NODE_ENV = 'production'` and `process.env.NODE_ENV = 'development'`

Here's a screenshot to show my console when building with `process.env.NODE_ENV = 'development'`:
![react-16-build](https://user-images.githubusercontent.com/4097374/29805478-ddab98b6-8c3e-11e7-9c1a-eb0f9e51e86f.png)

the red boxes show that the version of react is correct, and that the development script is included.

On the left is the rendered RJV component.

Can you help me replicate the warnings you're seeing?

**Comment by capaj** [comment]
I have this when I run a showcase for my library react-bound, here: https://github.com/capaj/react-bound/tree/master/e2e-test

just clone, npm i and `npm run showcase`
when you open the browser try editing some input

**Comment by mac-s-g** [comment]
okay, i did not test editing with React v16.  
thanks for the update.  I'll update after trying that.

**Comment by mac-s-g** [comment]
published to + react-json-view@1.12.1

**Comment by capaj** [comment]
@mac-s-g thanks, that worked splendidly
